File: amodeus/upstream/models/cycle_realization.py
from typing import Annotated

from pydantic import Field

from .base import IDModeusModel


class CycleRealization(IDModeusModel):
    name: str
    name_short: Annotated[str, Field(alias="nameShort")]
    code: str
    course_unit_realization_name_short: Annotated[
        str,
        Field(alias="courseUnitRealizationNameShort"),
    ]

File: src/main/kotlin/me/evgfilim1/amodeus/api/upstream/models/CycleRealization.kt
package me.evgfilim1.amodeus.api.upstream.models

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement
import kotlinx.serialization.json.JsonObject
import me.evgfilim1.amodeus.api.utils.UUID

@Serializable
data class CycleRealization(
    override val id: UUID,
    val name: String,
    val nameShort: String,
    val code: String,
    val courseUnitRealizationNameShort: String,
    override val _links: JsonObject,
) : IDModeusModel
